pandas/tests/scalar/timestamp/test_arithmetic.py

```diff
@@ -27,6 +28,33 @@ def test_overflow_offset(self):
         with pytest.raises(OverflowError):
             stamp - offset
 
+    def test_overflow_offset2(self):
```

Comment:
Hmm I see why you did the renaming now. Can you just combine this test and the one above. It looks like there a frew overlapping tests, so if there's anything new being tested just append it to the test above (with the issue number).

Reply:
* Let's also check error messages while we're at it here (i.e. `pytest.raises(OverflowError, match=...)`)
*  In the interests of greater modularity in the test base, I would prefer to split the tests into one checking on the expected errors, and one checking the expressions that shouldn't overflow.

Keep in mind that for large tests like these, a single failure prevents us from checking anything afterwards, so the smaller and modular the tests, the better.

Reply:
To add to what @gfyoung is saying we typically would create a test for working assertions and another one with `_raises` appended to the test name which tests the failures. You'll see this in other modules so probably a good standard to follow here

Reply:
Thanks all, sounds good.  I will plan on breaking up into one test with expressions that shouldn't overflow, and one that does (with _raises appended).  Note that the test that covers does-overflow will be a bit long. It seems important to test the specific cases from those issues though. But not sure if I should further break it up (have a look and let me know).